prime-router/docs/authz/authz-design.md

```diff
@@ -0,0 +1,279 @@
+| submit      | Able to submit reports                                                | ReportStream-Submit      |
+
+#### How to set up in Okta
+
+1. Create a scope
+<img src="img/okta-scope.png" width="600"/>
+<br><br>
+2. Create an Access Policy
+<img src="img/access-policy.png" width="600"/>
+<br><br>
+3. Add a Rule to the Policy. Ensure you have limited the scope to the group membership.
+<img src="img/access-policy-rule.png" width="600"/>
+<br><br>
+4. Add an additional rule to deny Access to users not in the group. Ensure the default scopes rule is first in priority.
+<img src="img/default-access-policy-rule.png" width="600"/>
+<br><br>
+
+### Organizations
+Organizations are set up as groups within Okta and will be included in the token. The combination of the scopes and 
+groups will allow us to make authorization decisions on most endpoints. Organizations are easily added to the token 
+for users via the UI. I will delve into how to set it up for senders later in the document.
+
+<img src="img/groups-claim.png" width="600"/>
+
+### Submit claim
+
+There will also be an optional `userSubmit` or `appSubmit` claim containing specific organizations AND optional sender. These values will 
+be dot seperated. If a sender is not included it will be assumed to be a wildcard and allow all senders under that 
+organization. Unfortunately the claim names have to seperated as to access the user profile and application user profile 
+require different expressions.
+
+examples:
+```
+"md-phd.full-elr": Only allow to send from md-phd as full-elr
+"ca-phd": Allow submitting from any sender under ca-phd
+```
+
+1. Update the default user to add an additional profile attribute
+<img src="img/submit-attr.png" width="600"/>
+<br><br>
+2. Add specific values to the user's profile that you want in the access token
+<img src="img/user-profile.png" width="600"/>
+<br><br>
+3. Add a claim referencing the user profile attribute
+<img src="img/user-submit.png" width="600"/>
+<br><br>
+
+
+### Claims example
+
+Given the following token claims scenario:
+```json
+{
+  "scp": [
+    "openid",
+    "email",
+    "org:read",
+    "submit"
+  ],
+  "org": [
+    "md-phd",
+    "ca-phd"
+  ],
+  "userSubmit": [
+    "md-phd.full-elr",
+    "ca-phd"  
+  ]  
+}
+```
+
+This particular user would be able to read information about `md-phd` and `ca-phd`. They would also be able to submit 
+reports only as `md-phd.full-elr` or for any sender under `ca-phd`. Note that the current implementation uses an 
+`organization` claim. This will continue to be present during the transition but will eventually be phased out.
+
+### Daily Data page special considerations
+
+For specific users within specific organizations, we want to allow them to access another organization's daily data 
+page. Since these are organizational level settings that are tightly coupled to Report Stream, we would store these 
+settings within the RS Postgres DB.
```

Comment:
As we said, this is probably fine for version 1.0 but it would be good to think on this and if there is a better way to manage and store org settings. For example, does Okta provide any facilities for this?